sw/device/lib/dif/dif_rv_core_ibex.c

```diff
@@ -0,0 +1,156 @@
+  uint32_t ibus_reg_en;
+  uint32_t dbus_addr_maching;
+  uint32_t dbus_addr_remap;
+  uint32_t dbus_addr_en;
+  uint32_t dbus_reg_en;
+} ibex_addr_translation_regs_t;
+
+static const ibex_addr_translation_regs_t
+    kRegsMap[kDifRvCoreIbexAddrTranslationSlotCount] = {
+        [kDifRvCoreIbexAddrTranslationSlot_0] =
+            {
+                .ibus_addr_maching =
+                    RV_CORE_IBEX_IBUS_ADDR_MATCHING_0_REG_OFFSET,
+                .ibus_addr_remap = RV_CORE_IBEX_IBUS_REMAP_ADDR_0_REG_OFFSET,
+                .ibus_addr_en = RV_CORE_IBEX_IBUS_ADDR_EN_0_REG_OFFSET,
+                .ibus_reg_en = RV_CORE_IBEX_IBUS_REGWEN_0_REG_OFFSET,
+                .dbus_addr_maching =
+                    RV_CORE_IBEX_DBUS_ADDR_MATCHING_0_REG_OFFSET,
+                .dbus_addr_remap = RV_CORE_IBEX_DBUS_REMAP_ADDR_0_REG_OFFSET,
+                .dbus_addr_en = RV_CORE_IBEX_DBUS_ADDR_EN_0_REG_OFFSET,
+                .dbus_reg_en = RV_CORE_IBEX_DBUS_REGWEN_0_REG_OFFSET,
+            },
+        [kDifRvCoreIbexAddrTranslationSlot_1] =
+            {
+                .ibus_addr_maching =
+                    RV_CORE_IBEX_IBUS_ADDR_MATCHING_1_REG_OFFSET,
+                .ibus_addr_remap = RV_CORE_IBEX_IBUS_REMAP_ADDR_1_REG_OFFSET,
+                .ibus_addr_en = RV_CORE_IBEX_IBUS_ADDR_EN_1_REG_OFFSET,
+                .ibus_reg_en = RV_CORE_IBEX_IBUS_REGWEN_1_REG_OFFSET,
+                .dbus_addr_maching =
+                    RV_CORE_IBEX_DBUS_ADDR_MATCHING_1_REG_OFFSET,
+                .dbus_addr_remap = RV_CORE_IBEX_DBUS_REMAP_ADDR_1_REG_OFFSET,
+                .dbus_addr_en = RV_CORE_IBEX_DBUS_ADDR_EN_1_REG_OFFSET,
+                .dbus_reg_en = RV_CORE_IBEX_DBUS_REGWEN_1_REG_OFFSET,
+            },
+};
+
+/**
+ * Convert the region address and size into a natural power of two address.
+ *
+ * @param addr Region start address.
+ * @param size region size.
+ * @return Napot address
+ */
+static uint32_t to_napot(uint32_t addr, size_t size) {
+  return addr | (size - 1) >> 1;
+}
+
+/**
+ * Split a natural power of two address into a start address and size.
+ *
+ * @param napot Address formated in NAPOT.
+ * @param size  Pointer to receive the region size.
+ * @return The region start address.
+ */
+static uint32_t from_napot(uint32_t napot, size_t *size) {
+  for (size_t i = 1; i < sizeof(uint32_t) * 8; ++i) {
+    uint32_t ref = (1u << i) - 1;
+    if ((napot & ref) == ref >> 1) {
+      *size = 1u << i;
+      break;
+    }
+  }
+  return napot & ~((*size - 1) >> 1);
+}
+
+dif_result_t dif_rv_core_ibex_configure_addr_translation(
+    const dif_rv_core_ibex_t *rv_core_ibex,
+    dif_rv_core_ibex_addr_translation_slot_t slot,
+    dif_rv_core_ibex_addr_translation_region_t region) {
+  if (rv_core_ibex == NULL || slot >= kDifRvCoreIbexAddrTranslationSlotCount) {
+    return kDifBadArg;
+  }
+
+  if (!bitfield_is_power_of_two32(region.dbus.size) ||
+      !bitfield_is_power_of_two32(region.ibus.size)) {
+    return kDifBadArg;
+  }
+
```

Comment:
Shouldn't we check if the REGWEN is set before proceeding, and return `kDifLocked` if it is not, indicating this configuration is locked out?

Reply:
Good catch! Fixed!